homeassistant/components/bluesound/media_player.py

```diff
@@ -64,6 +64,10 @@ async def async_setup_entry(
     async_add_entities: AddConfigEntryEntitiesCallback,
 ) -> None:
     """Set up the Bluesound entry."""
+    assert (
+        hasattr(config_entry, "runtime_data") and config_entry.runtime_data is not None
+    )
+
```

Comment:
The PR description and the fix mention that this could be unset or `None` and using `assert` here is hiding this case.
I don't understand why is the need to use `hasattr` for `runtime_data` and in general try to avoid using `has_attr` unless this is something which is not controlled by Home Assistant members and need to support backward compatibility.

Reply:
It can be unset if the config_entry has not been loaded. But in this case `async_setup_entry` in `__init__.py` must have already set it. The problem can only occure in `rebuild_bluesound_group`.

We need `hassattr` because the config_entry has no attribute `runtime_data`(even though if the typing says otherwise). And i know no other way to check for it's existence.

Reply:
The config entry can't be unloaded? Otherwise we wouldn't be at this stage?

Reply:
I think unloading has nothing to do with this. For join/unjoin of the media_players all other config_entries of this integration are fetched. And those in the state `SETUP_RETRY` do not have the `runtime_data` attribute. `SETUP_RETRY`happens often, because some users keep their amplifiers off when not in use.

I made it optional for that reason, but that was not enough. The attribute is not None. It is not there at all. And I had to add this assert as a consequence. I am pretty sure that it can never be unset at this point.